src/Parser/src/Parser.cpp

```diff
@@ -5,38 +5,150 @@
+            validVector = false;
+        }
+
+        validVector = validateParenthesis(equationVector);
+
+        std::pair < std::shared_ptr<std::vector<ExpressionUnit>>,bool> completedVector (equationVector, validVector);
+        return completedVector;
+    }
+
+    Expression Parser::breakDownEquation(const std::shared_ptr<std::vector<ExpressionUnit>> &equationVector)
     {
         Expression parsedExpression;
-        std::string expressionPiece = findNumber(fullEquation);
+        char importantOperator;
+        int parenthesisIndex;
+        char lastParenthesis = '(';
+        size_t startingPoint = 0;
+        size_t endingPoint = equationVector->size();
+        
+        std::this_thread::sleep_for(std::chrono::milliseconds(800));
 
-        parsedExpression.valid = validateFloat(expressionPiece);
+        for (size_t i = 0; i < equationVector->size(); i++) //find parenthesis that open then close
+        {
+            if (equationVector->at(i).operation == '(')
+            {
+                startingPoint = i;
+            }
+            if (equationVector->at(i).operation == ')' && lastParenthesis == '(')
+            {
+                endingPoint = i;
+                equationVector->erase(equationVector->begin() + startingPoint); //erase parenthesis
+                equationVector->erase(equationVector->begin() + (endingPoint - 1));
+                endingPoint = endingPoint - 2;
+                break;
+            }           
+            if (equationVector->at(i).operation == '(' || equationVector->at(i).operation == ')')
+            {
+                lastParenthesis = equationVector->at(i).operation;
+            }
+        }
 
-        if (parsedExpression.valid)
+        if ((startingPoint + 2) <= endingPoint) // if you enter something like "(1+)"
         {
-            parsedExpression.a = std::stof(expressionPiece);
-            parsedExpression.operation = fullEquation.substr(expressionPiece.length(), 1)[0];
-            parsedExpression.valid = validateOperator(parsedExpression.operation);
+            for (size_t i = startingPoint; i < endingPoint; i++) //find operator in order
+            {
+                if (equationVector->at(i).valid)
+                {
+                    parsedExpression.validExpression = true;
+                    if (equationVector->at(i).operation == '/' || equationVector->at(i).operation == '%' || equationVector->at(i).operation == 'x' || equationVector->at(i).operation == '*')
+                    {
+                        importantOperator = equationVector->at(i).operation;
+                        parsedExpression.placementIndex = i - 1;
+                        break;
+                    }
+                    if (equationVector->at(i).operation == '-' || equationVector->at(i).operation == '+')
+                    {
+                        if(importantOperator != '-' && importantOperator != '+')

```

Comment:
Bonus point: You are getting pretty deeply nested. If you have time try to refactor this, or hide with function calls